libs/application/core/src/lib/ApplicationTemplateHelper.ts

```diff
@@ -32,11 +41,14 @@ export class ApplicationTemplateHelper<
     this.template = template
   }
 
-  private initializeStateMachine(stateMachineContext?: TContext) {
+  private initializeStateMachine(
+    stateMachineContext?: TContext,
+    stateMachineOptions?: Partial<MachineOptions<TContext, TEvents>>,
+  ) {
     this.stateMachine = createApplicationMachine(
       this.application,
       this.template.stateMachineConfig,
-      this.template.stateMachineOptions,
+      merge(stateMachineOptions, this.template.stateMachineOptions),
```

Comment:
you can also just do {...stateMachineOptions, ...this.template.stateMachineOptions} right?